### PR TITLE
Spec flow

### PR DESCRIPTION
**Pull request by tarasiki**

### CODE REVIEW

**Review by dmatsev — APPROVED** [review]

**Review by SerhiiTarasiuk — APPROVED** [review]

**Review by eugenek7 — APPROVED** [review]

File: EasyRestProjectSpecflow/PageObjects/CurrentOrdersPage.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        private readonly IBrowserInteractions _browserInteractions;
+
+        public CurrentOrdersPage(IBrowserInteractions browserInteractions)
+        {
+            _browserInteractions = browserInteractions;
+        }
+        private IWebElement _currentOrdersButton => _browserInteractions.WaitAndReturnElement
+            (By.XPath("//span[contains(text(), 'Current Orders')]"));
+
+        private IWebElement _waitingforconfirm => _browserInteractions.WaitAndReturnElement
```
Comment (AhnessP) [inline EasyRestProjectSpecflow/PageObjects/CurrentOrdersPage.cs]: Better use camel case for " waitingforconfirm "element

File: EasyRestProjectSpecflow/PageObjects/OrderHistoryPage.cs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        {
+            _reorderButton.Click();
+        }
+
+        public void ClickOrderHistoryButton()
+        {
+            _orderHistoryButton.Click();
+        }
+
+        public string GetPopup()
```
Comment (AhnessP) [inline EasyRestProjectSpecflow/PageObjects/OrderHistoryPage.cs]: GetPopUp()

File: EasyRestProjectSpecflow/PageObjects/OrderHistoryPage.cs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        {
+            return _changedQuantityPopup.Text;
```
Comment (AhnessP) [inline EasyRestProjectSpecflow/PageObjects/OrderHistoryPage.cs]: _changeQuantityPopUp

File: EasyRestProjectSpecflow/Steps/MakeReorderStepDefinitions.cs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        public void ThenICheckThatOrderWithAppearsInWaitingToConfirm(string price)
+        {
+            _currentOrdersPage.ClickWaitingforconfirmButton();
+            var actualPrice = _currentOrdersPage.GetPrice();
+            var expectedPrice = price;
+            StringAssert.Contains(expectedPrice, actualPrice, "Problems with Order");
+        }
+
+        [Then(@"I check that '([^']*)' appears")]
+        public void ThenICheckThatAppears(string quantitypopup)
```
Comment (AhnessP) [inline EasyRestProjectSpecflow/Steps/MakeReorderStepDefinitions.cs:85]: quantityPopUp

File: SpecFlowProject/Pages/CurrentOrdersPage.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        
+        private readonly IBrowserInteractions _browserInteractions;
+
+        public CurrentOrdersPage(IBrowserInteractions browserInteractions)
+        {
+            _browserInteractions = browserInteractions;
+        }
+        private IWebElement _currentOrdersButton => _browserInteractions.WaitAndReturnElement(By.XPath("//span[contains(text(), 'Current Orders')]"));
+
+        private IWebElement _waitingforconfirm => _browserInteractions.WaitAndReturnElement(By.XPath("//span[contains(text(), 'Waiting for confirm')]"));
```
Comment (AhnessP) [inline SpecFlowProject/Pages/CurrentOrdersPage.cs:17]: _waitingForConfirm

File: SpecFlowProject/Pages/CurrentOrdersPage.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+        private IWebElement _firstOrder => _browserInteractions.WaitAndReturnElement(By.XPath("(//div[contains(@class, 'MuiButtonBase-root')]) [1]"));
+                
+        public void ClickCurrentOrdersButton()
+        {
+            _currentOrdersButton.Click();
+        }
+        public void ClickWaitingforconfirmButton()
```
Comment (AhnessP) [inline SpecFlowProject/Pages/CurrentOrdersPage.cs:25]: ClickWaitingForConfirmButton

**Review by AhnessP — APPROVED** [review]

File: EasyRestProjectSpecflow/PageObjects/CurrentOrdersPage.cs (hunk 1 above)
Comment (tarasiki) [reply EasyRestProjectSpecflow/PageObjects/CurrentOrdersPage.cs]: thank you for your comments)

**Pull request merged**